Source/portal.cpp

```diff
@@ -10,203 +10,174 @@ int WarpDropY[MAXPORTAL] = { 40, 40, 40, 40 };
 {
-	if ( portal[i].level == currlevel )
-		return 1;
-	else
-		return currlevel == 0;
+    if (portal[i].level == currlevel)
+        return TRUE;
+    else
+        return currlevel == 0;
 }
 
 void __fastcall RemovePortalMissile(int id)
 {
-	int i; // esi
-	int mi; // eax
-
-	for ( i = 0; i < nummissiles; ++i )
-	{
-		mi = missileactive[i];
-		if ( missile[mi]._mitype == MIS_TOWN && missile[mi]._misource == id )
-		{
-			dFlags[missile[mi]._mix][missile[mi]._miy] &= ~DFLAG_MISSILE;
-			dMissile[missile[mi]._mix][missile[mi]._miy] = 0;
-
-			if ( portal[id].level )
-				AddUnLight(missile[mi]._mlid);
-
-			DeleteMissile(mi, i);
-		}
-	}
+    int i;
+    int mi;
+
+    for (i = 0; i < nummissiles; ++i) {
+        mi = missileactive[i];
+        if (missile[mi]._mitype == MIS_TOWN && missile[mi]._misource == id) {
+            dFlags[missile[mi]._mix][missile[mi]._miy] &= ~DFLAG_MISSILE;
+            dMissile[missile[mi]._mix][missile[mi]._miy] = 0;
+
+            if (portal[id].level)
+                AddUnLight(missile[mi]._mlid);
+
+            DeleteMissile(mi, i);
+        }
+    }
 }
 
 void __fastcall SetCurrentPortal(int p)
 {
-	portalindex = p;
+    portalindex = p;
 }
 
 void __cdecl GetPortalLevel()
 {
-	if ( currlevel )
-	{
-		setlevel = 0;
-		currlevel = 0;
-		leveltype = 0;
-		plr[myplr].plrlevel = 0;
-	}
-	else
-	{
-		if ( portal[portalindex].setlvl )
-		{
-			setlevel = 1;
-			setlvlnum = portal[portalindex].level;
-		}
-		else
-		{
-			setlevel = 0;
-		}
-
-		currlevel = portal[portalindex].level;
-		leveltype = portal[portalindex].ltype;
-		plr[myplr].plrlevel = portal[portalindex].level;
-
-		if ( portalindex == myplr )
-		{
-			NetSendCmd(TRUE, CMD_DEACTIVATEPORTAL);
-			DeactivatePortal(portalindex);
-		}
-	}
+    if (currlevel) // go up to town
+    {
+        setlevel = 0;
+        currlevel = 0;
+        leveltype = 0;
+        plr[myplr].plrlevel = 0;
+        // BUGFIX? Portals last forever as long as you always travel to town...
+        return;
+    }
+    // else go to portal destination
+
+    if (portal[portalindex].setlvl) {
+        setlevel = 1;
+        setlvlnum = portal[portalindex].level;
+        plr[myplr].plrlevel = currlevel = setlvlnum;
+    } else {
+        setlevel = 0;
+        plr[myplr].plrlevel = currlevel = portal[portalindex].level;
+    }
+
+    leveltype = portal[portalindex].ltype;
+
+    if (portalindex == myplr) {
+        NetSendCmd(TRUE, CMD_DEACTIVATEPORTAL);
+        DeactivatePortal(portalindex);
+    }
 }
 // 5BB1ED: using guessed type char leveltype;
 // 5CCB10: using guessed type char setlvlnum;
 // 5CF31D: using guessed type char setlevel;
 
 void __cdecl GetPortalLvlPos()
 {
-	if ( currlevel )
-	{
-		ViewX = portal[portalindex].x;
-		ViewY = portal[portalindex].y;
-
-		if ( portalindex != myplr )
-		{
-			ViewX++;
-			ViewY++;
-		}
-	}
-	else
-	{
-		ViewX = WarpDropX[portalindex] + 1;
-		ViewY = WarpDropY[portalindex] + 1;
-	}
+    if (!currlevel) {

```

Comment:
currlevel == 0

Reply:
That's the exact same thing but less concise.

Reply:
WebKit style guide explicitly says not to use equality to test for zero/non-zero so I'd say this is consistent with our chosen style...